src/finbars/finbars.ts

```diff
@@ -0,0 +1,870 @@
+            sizeProp = this.oh.size,
+            maxScroll = Math.max(0, this.content[sizeProp] - containerRect[sizeProp]),
+            //scroll = Math.min(idx, maxScroll);
+            scroll = (idx - this._min) / (this._max - this._min) * maxScroll;
+        //console.log('scroll: ' + scroll);
+        this.content.style[this.oh.leading] = -scroll + 'px';
+    },
+
+    /**
+     * @summary Recalculate thumb position.
+     *
+     * @desc This method recalculates the thumb size and position. Call it once after inserting your scrollbar into the DOM, and repeatedly while resizing the scrollbar (which typically happens when the scrollbar's parent is resized by user.
+     *
+     * @param {Object} [options] - Optional options object.
+     * @param {number} [options.increment=this.increment] - Resets {@link FooBar#increment|increment} (see).
+     * @param {finbarStyles} [options.barStyles=this.barStyles] - (See type definition for details.) Scrollbar styles to be applied to the bar element.
+     * @param {number} [options.containerSize] - Defines the size of the container (visible part), when {@link FooBar#onChange|onChange} isn't {@link FooBar#scrollRealContent|scrollRealContent}.
+     *
+     * Only specify a `barStyles` object when you need to override stylesheet values. If provided, becomes the new default (`this.barStyles`), for use as a default on subsequent calls.
+     *
+     * It is generally the case that the scrollbar's new position is sufficiently described by the current styles. Therefore, it is unusual to need to provide a `barStyles` object on every call to `resize`.
+     *
+     * @returns {FinBar} Self for chaining.
+     * @memberOf FinBar.prototype
+     */
+    resize: function ({increment, barStyles}: {increment?: number, barStyles?: finbarStyles} = {}) {
+
+        var bar = this.bar;
+
+        if (!bar.parentNode) {
+            return; // not in DOM yet so nothing to do
+        }
+
+        var container = this.container || bar.parentElement,
+            containerRect = container.getBoundingClientRect();
+
+        this.style = this.barStyles = barStyles || this.barStyles;
+
+        // Bound to real content: Content was given but no onchange handler.
+        // Set up .onchange, .containerSize, and .increment.
+        // Note this only makes sense if your index unit is pixels.
+        if (this.content) {
+            if (!this.onchange) {
+                this.onchange = this.scrollRealContent;
+                this.contentSize = this.content[this.oh.size];
+                this._min = 0;
+                this._max = this.contentSize - 1;
+            }
+        }
+        if (this.onchange === this.scrollRealContent) {
+            this.containerSize = containerRect[this.oh.size];
+            this.increment = this.containerSize / (this.contentSize - this.containerSize) * (this._max - this._min);
+        } else {
+            this.increment = increment || this.increment;
```

Comment:
Key change here, this used to `this.containerSize = 1;`.
Now that step is skipped, and the containerSize should have been set before the resize.